### PR TITLE
Fix liquid template copy component 1337

### PR DESCRIPTION
**Pull request by JayKay24**
**Description**

This PR fixes [#1337 ](https://github.com/meshery/meshery.io/issues/1337)

**Notes for Reviewers**

**[Signed commits](https://docs.meshery.io/project/contributing#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Ghat0tkach** [comment]
hey @JayKay24 ,

Let's discuss it on the websites call.
Please add this as an agenda item in the meeting minutes, if you would :)

https://docs.google.com/document/d/1XczAHXVe2FIWPqiF57ospJ43zw5cZQ7ui8mn39v5EvA/edit#heading=h.lohhtewfwima

**Review by thisiskaransgit — COMMENTED** [review]
@JayKay24, thanks much, please take a look into the `footer.html` to make sure there's no overlapping script.

**Comment by JayKay24** [comment]
@thisiskaransgit Thanks for your review. The code in [footer.html](https://github.com/meshery/meshery.io/blob/a2002a46a2b3ab434f3c439a7aa26384b43639fc/_includes/footer.html#L75C1-L75C1) only has the javascript function(`resetCopyText`) and dependency(`clipboard.min.js`) needed to copy a piece of text. It is only defined in that file. I abstracted away the repeated HTML into a `copy-to-clipboard.html` template that still references the `resetCopyText` defined in `footer.html`.

**Review by thisiskaransgit — APPROVED** [review]
We are good to then, thanks @JayKay24 👍🏼

**Comment by JayKay24** [comment]
I've updated the branch to keep it mergable but will need a re-review @thisiskaransgit

**Review by thisiskaransgit — COMMENTED** [review]
@JayKay24, use of CDN, we need to replace it with a locally available file, otherwise the payload will not be improved. You can reference the implementation in docs.meshery.io if required. I overlooked it initially but we need this update

**Comment by JayKay24** [comment]
@thisiskaransgit does that mean that I integrate `npm` into the project? This would allow me to use the clipboard library currently loaded via CDN as a local npm module.

**Comment by thisiskaransgit** [comment]
> @thisiskaransgit does that mean that I integrate `npm` into the project? This would allow me to use the clipboard library currently loaded via CDN as a local npm module.

@JayKay24 declare the javascript logic in the code. the desired behavior should be that the class element can use the locally hosted code and not third-party module, which will a) reduce load b) filter the unused code that comes along c) opt out of depending on other services. This is what I intend to update for other sections too actually. 

take a look at here for reference:  https://github.com/meshery/meshery/blob/master/docs/assets/js/main.js

**Comment by Chadha93** [comment]
> > @thisiskaransgit does that mean that I integrate `npm` into the project? This would allow me to use the clipboard library currently loaded via CDN as a local npm module.
> 
> @JayKay24 declare the javascript logic in the code. the desired behavior should be that the class element can use the locally hosted code and not third-party module, which will a) reduce load b) filter the unused code that comes along c) opt out of depending on other services. This is what I intend to update for other sections too actually.
> 
> take a look at here for reference: [meshery/meshery@`master`/docs/assets/js/main.js](https://github.com/meshery/meshery/blob/master/docs/assets/js/main.js?rgh-link-date=2023-08-20T18%3A20%3A34Z)

@JayKay24, does this helps?

**Comment by JayKay24** [comment]
@Chadha93 i haven't had a chance to dive deep into the solution but from what I can tell is that I will wrap the clipboard logic inside an IIFE. Will look to address this tomorrow. I'll post any questions I have in this thread.

**Comment by JayKay24** [comment]
@thisiskaransgit The example you gave me also references the `clipboard.min.js` library via CDN in this file -> https://github.com/meshery/meshery/blob/master/docs/_layouts/default.html.

I don't understand what you mean when you say the payload will be improved. Which payload is this? Do you mean that what we want is to avoid the extra HTTP request of downloading the `clipboard.min.js` dependency on page load in order to achieve a faster loading rate by having the browser parse a  local `.js` file with clipboard logic?
 If this is the case, there's a way to use the dependency locally by downloading and [extracting the zip file](https://www.npmjs.com/package/clipboard) if we're not into package management. But if we do it this way, we won't have a way to keep it up to date easily.
The other option is to integrate the project with `npm` and reference `clipboard.js` as a node module.  This way when the project is built, the dependency will be included on page load without any extra HTTP request to the CDN.
Please help me to clarify this matter. Sorry If I'm not understanding it as quickly.
@Chadha93 please let me know your insights on this. What could I be missing?

File: _includes/help-modal.html [hunk 1]
```diff
@@ -111,36 +111,21 @@ <h3>Meshery CLI</h3>
                 <p class="code-box" style="height: 40px; line-height: 15px; color: white"><code style="position: absolute;">
                     mesheryctl pattern apply -f [file | URL]
                  </code></p>
-                  <!-- copy to clipboard -->
-                  <a class="btn tooltip" style="position: absolute; top:0.10rem; right: 0.5rem; font-size: 22px;" data-clipboard-target="#start-pattern"
-                    data-clipboard-text="mesheryctl pattern apply -f [file | URL]" onmouseout="resetCopyText(this)">
-                    <i class="far fa-copy"></i>
-                    <span class="tooltiptext" style="font-size: 15px; width: 140px; height: 40px; padding: 0; line-height: 40px; top: 2rem; left: -80px;">Copy to clipboard</span>
-                  </a>
+                  {% include copy-to-clipboard.html clipboard_target="#start-pattern" clipboard_text="mesheryctl pattern apply -f [file | URL]" %}
```
Comment (vishalvivekm) [inline _includes/help-modal.html:114]:
```suggestion
<!-- copy to clipboard -->
                  {% include copy-to-clipboard.html clipboard_target="#start-pattern" clipboard_text="mesheryctl pattern apply -f [file | URL]" %}
```

File: _includes/help-modal.html [hunk 2]
```diff
@@ -111,36 +111,21 @@ <h3>Meshery CLI</h3>
...
                 <p class="code-box" style="height: 40px; line-height: 15px; color: white"><code style="position: absolute;">
                     mesheryctl exp filter apply --file [GitHub Link]
                  </code></p>
-                  <!-- copy to clipboard -->
-                  <a class="btn tooltip" style="position: absolute; top:0.10rem; right: 0.5rem; font-size: 22px;" data-clipboard-target="#start-pattern"
-                    data-clipboard-text="mesheryctl exp filter apply --file [GitHub Link]" onmouseout="resetCopyText(this)">
-                    <i class="far fa-copy"></i>
-                    <span class="tooltiptext" style="font-size: 15px; width: 140px; height: 40px; padding: 0; line-height: 40px; top: 2rem; left: -80px;">Copy to clipboard</span>
-                  </a>
+                  {% include copy-to-clipboard.html clipboard_target="#start-pattern" clipboard_text="mesheryctl exp filter apply --file [GitHub Link]" %}
```
Comment (vishalvivekm) [inline _includes/help-modal.html:128]:
```suggestion
<!-- copy to clipboard -->
                  {% include copy-to-clipboard.html clipboard_target="#start-pattern" clipboard_text="mesheryctl exp filter apply --file [GitHub Link]" %} 
```

File: _includes/copy-to-clipboard.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<!-- copy to clipboard -->
+<a 
+  class="btn tooltip" 
+  data-clipboard-target="{{ clipboard_target }}"
```
Comment (vishalvivekm) [inline _includes/copy-to-clipboard.html:4]:
```suggestion
  data-clipboard-target="{{ include.clipboard_target }}"
```

File: _includes/copy-to-clipboard.html [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  data-clipboard-text="{{ clipboard_text }}"
```
Comment (vishalvivekm) [inline _includes/copy-to-clipboard.html:5]:
```suggestion
  data-clipboard-text="{{ include.clipboard_text }}"
```

**Review by vishalvivekm — APPROVED** [review]
Looks good to me.
Thanks @JayKay24 🚀

**Pull request merged**